For 2.x devtools, provide better messages when 1.x metadata is encountered

## Describe the bug

trying to explicitly specfify rhbq platform during creation gives this error:

```
qssnap create app -p com.quarkus.redhat:quarkus-universe-platform:1.11.6.Final-redhat-00001

Downloading from central: https://repo.maven.apache.org/maven2/com/quarkus/redhat/quarkus-universe-platform-quarkus-platform-descriptor/1.11.6.Final-redhat-00001/quarkus-universe-platform-quarkus-platform-descriptor-1.11.6.Final-redhat-00001-1.11.6.Final-redhat-00001.json
[ERROR] ❗ Unable to create project: Failed to resolve the Quarkus extension catalog
```

and I have settings.xml configured.

traditonal mvn create works:

```
mvn io.quarkus:quarkus-maven-plugin:1.11.6.Final-redhat-00001:create -DprojectGroupId=org.acme -DprojectArtifactId=getting-started -DplatformGroupId=com.redhat.quarkus -DplatformVersion=1.11.6.Final-redhat-00001 -DclassName="org.acme.quickstart.GreetingResource" -Dpath="/hello"
```

### Expected behavior
cli should know about repository locations defined by maven - not sure how that would work for gradle ? 

### Actual behavior

it seem to not consider maven settings at all.

doh - missed the flipped name ....but running with quarkus build from yesterday I get this:

```
 quarkus create app -p com.redhat.quarkus:quarkus-universe-bom:1.11.6.Final-redhat-00001 -a test234
[ERROR] ❗ Unable to create project: Failed to deserialize extension catalog /Users/max/.m2/repository/com/redhat/quarkus/quarkus-universe-bom-quarkus-platform-descriptor/1.11.6.Final-redhat-00001/quarkus-universe-bom-quarkus-platform-descriptor-1.11.6.Final-redhat-00001-1.11.6.Final-redhat-00001.json
```

are you able to run it to completion?

Something like this?
```
 % java -jar devtools/cli/target/quarkus-cli-999-SNAPSHOT-runner.jar create cli -o target/boo -P 1.13.7.Final
[ERROR] ❗  Unable to create project: The version 2 CLI can not be used with Quarkus 1.x projects.
The 1.x Quarkus CLI runner jar can be downloaded from Maven Central, e.g.
$ mvn org.apache.maven.plugins:maven-dependency-plugin:3.2.0:get \
       -Dtransitive=false -Dartifact=io.quarkus:quarkus-cli:1.13.7.Final:jar:runner
```

--- a/devtools/cli/src/main/java/io/quarkus/cli/CreateCli.java
+++ b/devtools/cli/src/main/java/io/quarkus/cli/CreateCli.java
@@ -78,7 +78,6 @@ public class CreateCli extends BaseCreateCommand {
             }
             return success ? CommandLine.ExitCode.OK : CommandLine.ExitCode.SOFTWARE;
         } catch (Exception e) {
-            output.error("Project creation failed, " + e.getMessage());
             return output.handleCommandException(e,
                     "Unable to create project: " + e.getMessage());
         }

--- a/devtools/cli/src/main/java/io/quarkus/cli/common/RegistryClientMixin.java
+++ b/devtools/cli/src/main/java/io/quarkus/cli/common/RegistryClientMixin.java
@@ -24,6 +24,10 @@ public class RegistryClientMixin {
     public QuarkusProject createQuarkusProject(Path projectRoot, TargetQuarkusVersionGroup targetVersion, BuildTool buildTool,
             OutputOptionMixin log) {
         ExtensionCatalog catalog = getExtensionCatalog(targetVersion, log);
+        if (catalog.getQuarkusCoreVersion().startsWith("1.")) {
+            throw new UnsupportedOperationException("The version 2 CLI can not be used with Quarkus 1.x projects.\n"
+                    + "Use the maven/gradle plugins when working with Quarkus 1.x projects.");
+        }
         return QuarkusProjectHelper.getProject(projectRoot, catalog, buildTool, log);
     }
 

--- a/docs/src/main/asciidoc/cli-tooling.adoc
+++ b/docs/src/main/asciidoc/cli-tooling.adoc
@@ -147,9 +147,8 @@ $ quarkus create cli --help
 ----
 
 [[specifying-quarkus-version]]
-[NOTE]
-.Specifying the Quarkus version
-====
+== Specifying the Quarkus version
+
 Both `quarkus create` and `quarkus extension list` allow you to explicitly specify a version of Quarkus in one of two ways:
 
 1. Specify a specific Platform Release BOM
@@ -172,7 +171,6 @@ A platform stream operates against the concept of a "registry". Each registry de
 Streams are identified using `platformKey:streamId` syntax. A specific stream can be specified using `-S platformKey:streamId`. When specifying a stream, empty segments will be replaced with _discovered_ defaults, based on stream resource resolution rules.
 +
 For `2.0.0.Final`, you must enable the registry client (`--registry-client`) explicitly to specify a stream. This will not be required in later releases.
-====
 
 == Dealing with extensions
 
@@ -313,3 +311,9 @@ Tests paused, press [r] to resume
 Note: Output will vary depending on the build tool your project is using (`maven`, `gradle`, or `jbang`).
 
 
+[[quarkus-version-compatibility]]
+[WARNING]
+.Compatibility with Quarkus 1.x
+====
+The version 2 Quarkus CLI can not be used with 1.x Quarkus projects or releases. Use the maven/gradle plugins when working with Quarkus 1.x projects.
+====

